### PR TITLE
v4-v5 upgrade scripts

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Programs to v4-v5 upgrade DAVL

### CODE REVIEW

File: upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import DAVL.V4 qualified as V4
+import Upgrade qualified as V4V5
+import DA.Foldable qualified as F
+
+data LedgerParties = LedgerParties {company : Party}
+
+main : LedgerParties -> Script ()
+main LedgerParties {company} = do
+  employeeRoles <- fmap snd <$> query @V4.EmployeeRole company
+  F.mapA_ createUpgradeProposal employeeRoles
```
Comment (hurryabit) [inline upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml]: With `forA_` you could avoid the auxiliary function by writing something along the lines of
```haskell
forA_ employeeRoles $ \V4.EmployeeRole {employee, company, ..} -> do
  ...
```

File: upgrade-v4-v5/scripts/upgrade-init/test-setup.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (hurryabit) [inline upgrade-v4-v5/scripts/upgrade-init/test-setup.json]: Can we please have only one copy of this file.

File: upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  where
+    createUpgradeProposal
+      : V4.EmployeeRole -> Script (Optional (ContractId V4V5.UpgradeProposal))
```
Comment (hurryabit) [inline upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml]: Why do we return an optional value if we don't use it?

File: upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    createUpgradeProposal V4.EmployeeRole {employee, company, ..} = do
+      existingProposal <- query @V4V5.UpgradeProposal employee
+      if null existingProposal
+      then do
+        Some <$> (submit company $ do
+           debug ("Creating upgrade proposal for " <> show employee)
+           createCmd (V4V5.UpgradeProposal employee company))
```
Comment (hurryabit) [inline upgrade-v4-v5/scripts/upgrade-init/daml/UpgradeInit.daml]:
```suggestion
           createCmd V4V5.UpgradeProposal{employee company})
```
or
```suggestion
           createCmd V4V5.UpgradeProposal with employee company)
```
Positional arguments of the same type are just scary. It's so easy to put them the wrong way around.

File: upgrade-v4-v5/scripts/upgrade-accept/daml/UpgradeAccept.daml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import Upgrade qualified as V4V5
+import DA.Foldable qualified as F
+
+data LedgerParties = LedgerParties {company : Party}
+
+main : LedgerParties -> Script ()
+main LedgerParties {company} = do
+  upgradeProposals <- query @V4V5.UpgradeProposal company
+  F.mapA_ acceptUpgradeProposal upgradeProposals
```
Comment (hurryabit) [inline upgrade-v4-v5/scripts/upgrade-accept/daml/UpgradeAccept.daml]: Same comment as below regarding `forA_`.

File: upgrade-v4-v5/scripts/upgrade-accept/daml.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+sdk-version: 0.13.51
+name: upgrade-accept
```
[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply upgrade-v4-v5/scripts/upgrade-accept/daml.yaml]: let's come back to the final layout when everything is working.

**Pull request merged**